gateway/slack.go

```diff
@@ -171,20 +172,84 @@ func (sc *SlackClient) regenerateReverseMappings() {
 
 	sc.nickToUserMap = make(map[string]string)
 	for _, user := range sc.userInfo {
+		if user == nil {
+			continue
+		}
 		sc.nickToUserMap[user.Nick] = user.SlackID
 	}
 
 	sc.channelNameToIDMap = make(map[string]string)
 	for _, channel := range sc.channelInfo {
+		if channel == nil {
+			continue
+		}
 		sc.channelNameToIDMap[channel.Name] = channel.SlackID
 	}
 
 	sc.userIDToDMIDMap = make(map[string]string)
 	for dmID, user := range sc.dmInfo {
+		if user == nil {
+			continue
+		}
 		sc.userIDToDMIDMap[user.SlackID] = dmID
 	}
 }
 
+// Clean up our mappings if necessary
+func (sc *SlackClient) cleanupMappings() {
```

Comment:
maybe we should do the whole thing in one go, and not lock and unlock for each stage, so other goroutines don't get potentially stale data